{"metadata": {"kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat_minor": 5, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "## Data prep for MR", "metadata": {}}, {"cell_type": "markdown", "source": "#### Thank you Storm et al: https://www.nature.com/articles/s41467-021-26280-1 and Mishra et al: https://www.researchgate.net/publication/357596053_Stroke_genetics_informs_drug_discovery_and_risk_prediction_across_ancestries", "metadata": {}}, {"cell_type": "code", "source": "import pandas as pd\nimport numpy as np", "metadata": {}, "execution_count": 1, "outputs": []}, {"cell_type": "code", "source": "%%bash\ndx download ALS_MR/eQTL_data/eqtl_data_eqtlgen/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz", "metadata": {}, "execution_count": 4, "outputs": []}, {"cell_type": "code", "source": "eqtlgen = pd.read_csv('2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz', delim_whitespace=True)\neqtlgen.head()", "metadata": {}, "execution_count": 6, "outputs": [{"execution_count": 6, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>GenePos</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs12230244</td>\n", "      <td>12</td>\n", "      <td>10117369</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>200.7534</td>\n", "      <td>ENSG00000172322</td>\n", "      <td>CLEC12A</td>\n", "      <td>12</td>\n", "      <td>10126104</td>\n", "      <td>34</td>\n", "      <td>30596</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs12229020</td>\n", "      <td>12</td>\n", "      <td>10117683</td>\n", "      <td>G</td>\n", "      <td>C</td>\n", "      <td>200.6568</td>\n", "      <td>ENSG00000172322</td>\n", "      <td>CLEC12A</td>\n", "      <td>12</td>\n", "      <td>10126104</td>\n", "      <td>34</td>\n", "      <td>30596</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs61913527</td>\n", "      <td>12</td>\n", "      <td>10116198</td>\n", "      <td>T</td>\n", "      <td>C</td>\n", "      <td>200.2654</td>\n", "      <td>ENSG00000172322</td>\n", "      <td>CLEC12A</td>\n", "      <td>12</td>\n", "      <td>10126104</td>\n", "      <td>34</td>\n", "      <td>30598</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs2594103</td>\n", "      <td>12</td>\n", "      <td>10115428</td>\n", "      <td>T</td>\n", "      <td>C</td>\n", "      <td>200.0420</td>\n", "      <td>ENSG00000172322</td>\n", "      <td>CLEC12A</td>\n", "      <td>12</td>\n", "      <td>10126104</td>\n", "      <td>34</td>\n", "      <td>30598</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs12231833</td>\n", "      <td>12</td>\n", "      <td>10118428</td>\n", "      <td>A</td>\n", "      <td>G</td>\n", "      <td>199.9508</td>\n", "      <td>ENSG00000172322</td>\n", "      <td>CLEC12A</td>\n", "      <td>12</td>\n", "      <td>10126104</td>\n", "      <td>34</td>\n", "      <td>30592</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["          Pvalue         SNP  SNPChr    SNPPos AssessedAllele OtherAllele  \\\n", "0  3.271700e-310  rs12230244      12  10117369              T           A   \n", "1  3.271700e-310  rs12229020      12  10117683              G           C   \n", "2  3.271700e-310  rs61913527      12  10116198              T           C   \n", "3  3.271700e-310   rs2594103      12  10115428              T           C   \n", "4  3.271700e-310  rs12231833      12  10118428              A           G   \n", "\n", "     Zscore             Gene GeneSymbol  GeneChr   GenePos  NrCohorts  \\\n", "0  200.7534  ENSG00000172322    CLEC12A       12  10126104         34   \n", "1  200.6568  ENSG00000172322    CLEC12A       12  10126104         34   \n", "2  200.2654  ENSG00000172322    CLEC12A       12  10126104         34   \n", "3  200.0420  ENSG00000172322    CLEC12A       12  10126104         34   \n", "4  199.9508  ENSG00000172322    CLEC12A       12  10126104         34   \n", "\n", "   NrSamples  FDR    BonferroniP  \n", "0      30596  0.0  4.166200e-302  \n", "1      30596  0.0  4.166200e-302  \n", "2      30598  0.0  4.166200e-302  \n", "3      30598  0.0  4.166200e-302  \n", "4      30592  0.0  4.166200e-302  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "%%bash\ndx download ALS_MR/eQTL_data/eqtl_data_eqtlgen/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt.gz", "metadata": {}, "execution_count": 7, "outputs": []}, {"cell_type": "code", "source": "allele = pd.read_csv('2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt.gz', delim_whitespace=True)\nallele.head()", "metadata": {}, "execution_count": 8, "outputs": [{"execution_count": 8, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNP</th>\n", "      <th>hg19_chr</th>\n", "      <th>hg19_pos</th>\n", "      <th>AlleleA</th>\n", "      <th>AlleleB</th>\n", "      <th>allA_total</th>\n", "      <th>allAB_total</th>\n", "      <th>allB_total</th>\n", "      <th>AlleleB_all</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>rs140337953</td>\n", "      <td>1</td>\n", "      <td>30923</td>\n", "      <td>T</td>\n", "      <td>G</td>\n", "      <td>71.0</td>\n", "      <td>10.0</td>\n", "      <td>1.0</td>\n", "      <td>0.073171</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>rs146477069</td>\n", "      <td>1</td>\n", "      <td>54421</td>\n", "      <td>A</td>\n", "      <td>G</td>\n", "      <td>75.0</td>\n", "      <td>6.0</td>\n", "      <td>0.0</td>\n", "      <td>0.037037</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>rs141149254</td>\n", "      <td>1</td>\n", "      <td>54490</td>\n", "      <td>G</td>\n", "      <td>A</td>\n", "      <td>381.0</td>\n", "      <td>122.0</td>\n", "      <td>5.0</td>\n", "      <td>0.129921</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>rs2462492</td>\n", "      <td>1</td>\n", "      <td>54676</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>54.0</td>\n", "      <td>64.0</td>\n", "      <td>29.0</td>\n", "      <td>0.414966</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>rs143174675</td>\n", "      <td>1</td>\n", "      <td>54753</td>\n", "      <td>T</td>\n", "      <td>G</td>\n", "      <td>73.0</td>\n", "      <td>8.0</td>\n", "      <td>1.0</td>\n", "      <td>0.060976</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["           SNP  hg19_chr  hg19_pos AlleleA AlleleB  allA_total  allAB_total  \\\n", "0  rs140337953         1     30923       T       G        71.0         10.0   \n", "1  rs146477069         1     54421       A       G        75.0          6.0   \n", "2  rs141149254         1     54490       G       A       381.0        122.0   \n", "3    rs2462492         1     54676       C       T        54.0         64.0   \n", "4  rs143174675         1     54753       T       G        73.0          8.0   \n", "\n", "   allB_total  AlleleB_all  \n", "0         1.0     0.073171  \n", "1         0.0     0.037037  \n", "2         5.0     0.129921  \n", "3        29.0     0.414966  \n", "4         1.0     0.060976  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "%%bash\ndx download ALS_MR/druggable_genome.csv", "metadata": {}, "execution_count": 9, "outputs": []}, {"cell_type": "code", "source": "dg = pd.read_csv('druggable_genome.csv')\ndg.head()", "metadata": {}, "execution_count": 12, "outputs": [{"execution_count": 12, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>druggability_tier</th>\n", "      <th>hgnc_names</th>\n", "      <th>chr_b37</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>strand</th>\n", "      <th>description</th>\n", "      <th>no_of_gwas_regions</th>\n", "      <th>small_mol_druggable</th>\n", "      <th>bio_druggable</th>\n", "      <th>adme_gene</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000000938</td>\n", "      <td>Tier 1</td>\n", "      <td>FGR</td>\n", "      <td>1</td>\n", "      <td>27938575</td>\n", "      <td>27961788</td>\n", "      <td>-1</td>\n", "      <td>feline Gardner-Rasheed sarcoma viral oncogene ...</td>\n", "      <td>0</td>\n", "      <td>Y</td>\n", "      <td>N</td>\n", "      <td>N</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000001626</td>\n", "      <td>Tier 1</td>\n", "      <td>CFTR</td>\n", "      <td>7</td>\n", "      <td>117105838</td>\n", "      <td>117356025</td>\n", "      <td>1</td>\n", "      <td>cystic fibrosis transmembrane conductance regu...</td>\n", "      <td>0</td>\n", "      <td>Y</td>\n", "      <td>N</td>\n", "      <td>Y</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>ENSG00000001630</td>\n", "      <td>Tier 1</td>\n", "      <td>CYP51A1</td>\n", "      <td>7</td>\n", "      <td>91741465</td>\n", "      <td>91772266</td>\n", "      <td>-1</td>\n", "      <td>cytochrome P450, family 51, subfamily A, polyp...</td>\n", "      <td>0</td>\n", "      <td>Y</td>\n", "      <td>N</td>\n", "      <td>Y</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>ENSG00000002549</td>\n", "      <td>Tier 1</td>\n", "      <td>LAP3</td>\n", "      <td>4</td>\n", "      <td>17578815</td>\n", "      <td>17609595</td>\n", "      <td>1</td>\n", "      <td>leucine aminopeptidase 3 [Source:HGNC Symbol;A...</td>\n", "      <td>0</td>\n", "      <td>Y</td>\n", "      <td>N</td>\n", "      <td>N</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>ENSG00000004468</td>\n", "      <td>Tier 1</td>\n", "      <td>CD38</td>\n", "      <td>4</td>\n", "      <td>15779898</td>\n", "      <td>15854853</td>\n", "      <td>1</td>\n", "      <td>CD38 molecule [Source:HGNC Symbol;Acc:1667]</td>\n", "      <td>0</td>\n", "      <td>N</td>\n", "      <td>Y</td>\n", "      <td>N</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   ensembl_gene_id druggability_tier hgnc_names chr_b37  start_b37    end_b37  \\\n", "0  ENSG00000000938            Tier 1        FGR       1   27938575   27961788   \n", "1  ENSG00000001626            Tier 1       CFTR       7  117105838  117356025   \n", "2  ENSG00000001630            Tier 1    CYP51A1       7   91741465   91772266   \n", "3  ENSG00000002549            Tier 1       LAP3       4   17578815   17609595   \n", "4  ENSG00000004468            Tier 1       CD38       4   15779898   15854853   \n", "\n", "   strand                                        description  \\\n", "0      -1  feline Gardner-Rasheed sarcoma viral oncogene ...   \n", "1       1  cystic fibrosis transmembrane conductance regu...   \n", "2      -1  cytochrome P450, family 51, subfamily A, polyp...   \n", "3       1  leucine aminopeptidase 3 [Source:HGNC Symbol;A...   \n", "4       1        CD38 molecule [Source:HGNC Symbol;Acc:1667]   \n", "\n", "   no_of_gwas_regions small_mol_druggable bio_druggable adme_gene  \n", "0                   0                   Y             N         N  \n", "1                   0                   Y             N         Y  \n", "2                   0                   Y             N         Y  \n", "3                   0                   Y             N         N  \n", "4                   0                   N             Y         N  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "dg_tier1 = dg[dg['druggability_tier']=='Tier 1']\ndg_t1_nox = dg_tier1[dg_tier1['chr_b37'] != 'X']\ndg_t1_nox.info()", "metadata": {}, "execution_count": 13, "outputs": [{"name": "stdout", "output_type": "stream", "text": "<class 'pandas.core.frame.DataFrame'>\n\nInt64Index: 1377 entries, 0 to 1426\n\nData columns (total 12 columns):\n\nensembl_gene_id        1377 non-null object\n\ndruggability_tier      1377 non-null object\n\nhgnc_names             1375 non-null object\n\nchr_b37                1377 non-null object\n\nstart_b37              1377 non-null int64\n\nend_b37                1377 non-null int64\n\nstrand                 1377 non-null int64\n\ndescription            1377 non-null object\n\nno_of_gwas_regions     1377 non-null int64\n\nsmall_mol_druggable    1377 non-null object\n\nbio_druggable          1377 non-null object\n\nadme_gene              1377 non-null object\n\ndtypes: int64(4), object(8)\n\nmemory usage: 139.9+ KB\n"}]}, {"cell_type": "code", "source": "pos = dg_t1_nox[['ensembl_gene_id', 'start_b37', 'end_b37']]\npos.head()", "metadata": {}, "execution_count": 14, "outputs": [{"execution_count": 14, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000000938</td>\n", "      <td>27938575</td>\n", "      <td>27961788</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000001626</td>\n", "      <td>117105838</td>\n", "      <td>117356025</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>ENSG00000001630</td>\n", "      <td>91741465</td>\n", "      <td>91772266</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>ENSG00000002549</td>\n", "      <td>17578815</td>\n", "      <td>17609595</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>ENSG00000004468</td>\n", "      <td>15779898</td>\n", "      <td>15854853</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   ensembl_gene_id  start_b37    end_b37\n", "0  ENSG00000000938   27938575   27961788\n", "1  ENSG00000001626  117105838  117356025\n", "2  ENSG00000001630   91741465   91772266\n", "3  ENSG00000002549   17578815   17609595\n", "4  ENSG00000004468   15779898   15854853"]}, "metadata": {}}]}, {"cell_type": "code", "source": "gene_list = dg_t1_nox['ensembl_gene_id']\neqtl_filter= eqtlgen[eqtlgen['Gene'].isin(gene_list)]\neqtl_filter.info()", "metadata": {}, "execution_count": 16, "outputs": [{"name": "stdout", "output_type": "stream", "text": "<class 'pandas.core.frame.DataFrame'>\n\nInt64Index: 668141 entries, 13 to 10507655\n\nData columns (total 15 columns):\n\nPvalue            668141 non-null float64\n\nSNP               668141 non-null object\n\nSNPChr            668141 non-null int64\n\nSNPPos            668141 non-null int64\n\nAssessedAllele    668141 non-null object\n\nOtherAllele       668141 non-null object\n\nZscore            668141 non-null float64\n\nGene              668141 non-null object\n\nGeneSymbol        668141 non-null object\n\nGeneChr           668141 non-null int64\n\nGenePos           668141 non-null int64\n\nNrCohorts         668141 non-null int64\n\nNrSamples         668141 non-null int64\n\nFDR               668141 non-null float64\n\nBonferroniP       668141 non-null float64\n\ndtypes: float64(4), int64(6), object(5)\n\nmemory usage: 81.6+ MB\n"}]}, {"cell_type": "code", "source": "merge1 = pd.merge(eqtl_filter, pos, left_on='Gene', right_on='ensembl_gene_id', how='outer')\nmerge1.head()", "metadata": {}, "execution_count": 17, "outputs": [{"execution_count": 17, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>GenePos</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs964611</td>\n", "      <td>15.0</td>\n", "      <td>48597514.0</td>\n", "      <td>A</td>\n", "      <td>C</td>\n", "      <td>193.7198</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs61248772</td>\n", "      <td>15.0</td>\n", "      <td>48598726.0</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>192.9769</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs74011998</td>\n", "      <td>15.0</td>\n", "      <td>48596713.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.9409</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13753.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7168752</td>\n", "      <td>15.0</td>\n", "      <td>48600780.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.7138</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs79040993</td>\n", "      <td>15.0</td>\n", "      <td>48602146.0</td>\n", "      <td>T</td>\n", "      <td>G</td>\n", "      <td>192.4130</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13754.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["          Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "0  3.271700e-310    rs964611    15.0  48597514.0              A           C   \n", "1  3.271700e-310  rs61248772    15.0  48598726.0              T           A   \n", "2  3.271700e-310  rs74011998    15.0  48596713.0              C           T   \n", "3  3.271700e-310   rs7168752    15.0  48600780.0              C           T   \n", "4  3.271700e-310  rs79040993    15.0  48602146.0              T           G   \n", "\n", "     Zscore             Gene GeneSymbol  GeneChr     GenePos  NrCohorts  \\\n", "0  193.7198  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "1  192.9769  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "2  192.9409  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "3  192.7138  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "4  192.4130  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "\n", "   NrSamples  FDR    BonferroniP  ensembl_gene_id  start_b37   end_b37  \n", "0    13755.0  0.0  4.166200e-302  ENSG00000074803   48483861  48596275  \n", "1    13755.0  0.0  4.166200e-302  ENSG00000074803   48483861  48596275  \n", "2    13753.0  0.0  4.166200e-302  ENSG00000074803   48483861  48596275  \n", "3    13755.0  0.0  4.166200e-302  ENSG00000074803   48483861  48596275  \n", "4    13754.0  0.0  4.166200e-302  ENSG00000074803   48483861  48596275  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "cols = merge1.columns\ngene = merge1['GeneSymbol'].unique()\n\neqtl_5kb = pd.DataFrame(columns=cols)\n\nmerge1['SNPPos'] = pd.to_numeric(merge1['SNPPos'])\nmerge1['start_b37'] = pd.to_numeric(merge1['start_b37'])\nmerge1['end_b37'] = pd.to_numeric(merge1['end_b37'])\n\n\nfor i in gene:\n\n    gene_fil = i\n    dat1 = merge1[merge1['GeneSymbol']==gene_fil]\n    dat1 = dat1[(dat1['SNPPos'] > dat1['start_b37']-5000) & (dat1['SNPPos'] < dat1['end_b37']+5000)]\n    \n    eqtl_5kb = eqtl_5kb.append(dat1)", "metadata": {}, "execution_count": 25, "outputs": []}, {"cell_type": "code", "source": "eqtl_5kb.info()", "metadata": {}, "execution_count": 28, "outputs": [{"name": "stdout", "output_type": "stream", "text": "<class 'pandas.core.frame.DataFrame'>\n\nInt64Index: 96361 entries, 0 to 668137\n\nData columns (total 18 columns):\n\nPvalue             96361 non-null float64\n\nSNP                96361 non-null object\n\nSNPChr             96361 non-null float64\n\nSNPPos             96361 non-null float64\n\nAssessedAllele     96361 non-null object\n\nOtherAllele        96361 non-null object\n\nZscore             96361 non-null float64\n\nGene               96361 non-null object\n\nGeneSymbol         96361 non-null object\n\nGeneChr            96361 non-null float64\n\nGenePos            96361 non-null float64\n\nNrCohorts          96361 non-null float64\n\nNrSamples          96361 non-null float64\n\nFDR                96361 non-null float64\n\nBonferroniP        96361 non-null float64\n\nensembl_gene_id    96361 non-null object\n\nstart_b37          96361 non-null object\n\nend_b37            96361 non-null object\n\ndtypes: float64(10), object(8)\n\nmemory usage: 14.0+ MB\n"}]}, {"cell_type": "code", "source": "eqtl_5kb.head()", "metadata": {}, "execution_count": 29, "outputs": [{"execution_count": 29, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>GenePos</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs964611</td>\n", "      <td>15.0</td>\n", "      <td>48597514.0</td>\n", "      <td>A</td>\n", "      <td>C</td>\n", "      <td>193.7198</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs61248772</td>\n", "      <td>15.0</td>\n", "      <td>48598726.0</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>192.9769</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs74011998</td>\n", "      <td>15.0</td>\n", "      <td>48596713.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.9409</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13753.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7168752</td>\n", "      <td>15.0</td>\n", "      <td>48600780.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.7138</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "    <tr>\n", "      <th>11</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7162936</td>\n", "      <td>15.0</td>\n", "      <td>48594608.0</td>\n", "      <td>C</td>\n", "      <td>G</td>\n", "      <td>159.4042</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>6.0</td>\n", "      <td>9671.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["           Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "0   3.271700e-310    rs964611    15.0  48597514.0              A           C   \n", "1   3.271700e-310  rs61248772    15.0  48598726.0              T           A   \n", "2   3.271700e-310  rs74011998    15.0  48596713.0              C           T   \n", "3   3.271700e-310   rs7168752    15.0  48600780.0              C           T   \n", "11  3.271700e-310   rs7162936    15.0  48594608.0              C           G   \n", "\n", "      Zscore             Gene GeneSymbol  GeneChr     GenePos  NrCohorts  \\\n", "0   193.7198  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "1   192.9769  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "2   192.9409  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "3   192.7138  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "11  159.4042  ENSG00000074803    SLC12A1     15.0  48540068.0        6.0   \n", "\n", "    NrSamples  FDR    BonferroniP  ensembl_gene_id start_b37   end_b37  \n", "0     13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275  \n", "1     13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275  \n", "2     13753.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275  \n", "3     13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275  \n", "11     9671.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "allele_info = allele[['SNP','AlleleB', 'AlleleB_all']]", "metadata": {}, "execution_count": 30, "outputs": []}, {"cell_type": "code", "source": "merge2 = pd.merge(eqtl_5kb, allele_info, on='SNP' )\nmerge2.head()", "metadata": {}, "execution_count": 31, "outputs": [{"execution_count": 31, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>GenePos</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>AlleleB</th>\n", "      <th>AlleleB_all</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs964611</td>\n", "      <td>15.0</td>\n", "      <td>48597514.0</td>\n", "      <td>A</td>\n", "      <td>C</td>\n", "      <td>193.7198</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>A</td>\n", "      <td>0.156582</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs61248772</td>\n", "      <td>15.0</td>\n", "      <td>48598726.0</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>192.9769</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.156809</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs74011998</td>\n", "      <td>15.0</td>\n", "      <td>48596713.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.9409</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13753.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156557</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7168752</td>\n", "      <td>15.0</td>\n", "      <td>48600780.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.7138</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156582</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7162936</td>\n", "      <td>15.0</td>\n", "      <td>48594608.0</td>\n", "      <td>C</td>\n", "      <td>G</td>\n", "      <td>159.4042</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>48540068.0</td>\n", "      <td>6.0</td>\n", "      <td>9671.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156247</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["          Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "0  3.271700e-310    rs964611    15.0  48597514.0              A           C   \n", "1  3.271700e-310  rs61248772    15.0  48598726.0              T           A   \n", "2  3.271700e-310  rs74011998    15.0  48596713.0              C           T   \n", "3  3.271700e-310   rs7168752    15.0  48600780.0              C           T   \n", "4  3.271700e-310   rs7162936    15.0  48594608.0              C           G   \n", "\n", "     Zscore             Gene GeneSymbol  GeneChr     GenePos  NrCohorts  \\\n", "0  193.7198  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "1  192.9769  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "2  192.9409  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "3  192.7138  ENSG00000074803    SLC12A1     15.0  48540068.0       15.0   \n", "4  159.4042  ENSG00000074803    SLC12A1     15.0  48540068.0        6.0   \n", "\n", "   NrSamples  FDR    BonferroniP  ensembl_gene_id start_b37   end_b37 AlleleB  \\\n", "0    13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       A   \n", "1    13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       T   \n", "2    13753.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "3    13755.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "4     9671.0  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "\n", "   AlleleB_all  \n", "0     0.156582  \n", "1     0.156809  \n", "2     0.156557  \n", "3     0.156582  \n", "4     0.156247  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "flip = merge2[merge2['AssessedAllele'] != merge2['AlleleB']]\nflip.head()", "metadata": {}, "execution_count": 60, "outputs": [{"execution_count": 60, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>...</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>AlleleB</th>\n", "      <th>AlleleB_all</th>\n", "      <th>new_f</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>5</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs11855410</td>\n", "      <td>15.0</td>\n", "      <td>48593854.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>122.8678</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13742.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.817454</td>\n", "      <td>0.817454</td>\n", "    </tr>\n", "    <tr>\n", "      <th>10</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs72623972</td>\n", "      <td>15.0</td>\n", "      <td>48586994.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>115.8559</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.819253</td>\n", "      <td>0.819253</td>\n", "    </tr>\n", "    <tr>\n", "      <th>11</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs8026751</td>\n", "      <td>15.0</td>\n", "      <td>48582845.0</td>\n", "      <td>A</td>\n", "      <td>G</td>\n", "      <td>115.6258</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>G</td>\n", "      <td>0.819053</td>\n", "      <td>0.819053</td>\n", "    </tr>\n", "    <tr>\n", "      <th>12</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7174204</td>\n", "      <td>15.0</td>\n", "      <td>48579146.0</td>\n", "      <td>G</td>\n", "      <td>C</td>\n", "      <td>115.1496</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.815279</td>\n", "      <td>0.815279</td>\n", "    </tr>\n", "    <tr>\n", "      <th>15</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs60820695</td>\n", "      <td>15.0</td>\n", "      <td>48584883.0</td>\n", "      <td>T</td>\n", "      <td>C</td>\n", "      <td>110.9321</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.852457</td>\n", "      <td>0.852457</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>5 rows \u00d7 21 columns</p>\n", "</div>"], "text/plain": ["           Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "5   3.271700e-310  rs11855410    15.0  48593854.0              C           T   \n", "10  3.271700e-310  rs72623972    15.0  48586994.0              C           T   \n", "11  3.271700e-310   rs8026751    15.0  48582845.0              A           G   \n", "12  3.271700e-310   rs7174204    15.0  48579146.0              G           C   \n", "15  3.271700e-310  rs60820695    15.0  48584883.0              T           C   \n", "\n", "      Zscore             Gene GeneSymbol  GeneChr  ...  NrCohorts  NrSamples  \\\n", "5   122.8678  ENSG00000074803    SLC12A1     15.0  ...       15.0    13742.0   \n", "10  115.8559  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "11  115.6258  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "12  115.1496  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "15  110.9321  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "\n", "    FDR    BonferroniP  ensembl_gene_id start_b37   end_b37 AlleleB  \\\n", "5   0.0  4.166200e-302  ENSG00000074803  48483861  48596275       T   \n", "10  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       T   \n", "11  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       G   \n", "12  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "15  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "\n", "   AlleleB_all     new_f  \n", "5     0.817454  0.817454  \n", "10    0.819253  0.819253  \n", "11    0.819053  0.819053  \n", "12    0.815279  0.815279  \n", "15    0.852457  0.852457  \n", "\n", "[5 rows x 21 columns]"]}, "metadata": {}}]}, {"cell_type": "code", "source": "flip['new_f'] = 1 - flip['AlleleB_all']\nflip.head()", "metadata": {}, "execution_count": 61, "outputs": [{"name": "stderr", "output_type": "stream", "text": "/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n\nA value is trying to be set on a copy of a slice from a DataFrame.\n\nTry using .loc[row_indexer,col_indexer] = value instead\n\n\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  \"\"\"Entry point for launching an IPython kernel.\n"}, {"execution_count": 61, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>...</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>AlleleB</th>\n", "      <th>AlleleB_all</th>\n", "      <th>new_f</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>5</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs11855410</td>\n", "      <td>15.0</td>\n", "      <td>48593854.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>122.8678</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13742.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.817454</td>\n", "      <td>0.182546</td>\n", "    </tr>\n", "    <tr>\n", "      <th>10</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs72623972</td>\n", "      <td>15.0</td>\n", "      <td>48586994.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>115.8559</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.819253</td>\n", "      <td>0.180747</td>\n", "    </tr>\n", "    <tr>\n", "      <th>11</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs8026751</td>\n", "      <td>15.0</td>\n", "      <td>48582845.0</td>\n", "      <td>A</td>\n", "      <td>G</td>\n", "      <td>115.6258</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>G</td>\n", "      <td>0.819053</td>\n", "      <td>0.180947</td>\n", "    </tr>\n", "    <tr>\n", "      <th>12</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7174204</td>\n", "      <td>15.0</td>\n", "      <td>48579146.0</td>\n", "      <td>G</td>\n", "      <td>C</td>\n", "      <td>115.1496</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.815279</td>\n", "      <td>0.184721</td>\n", "    </tr>\n", "    <tr>\n", "      <th>15</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs60820695</td>\n", "      <td>15.0</td>\n", "      <td>48584883.0</td>\n", "      <td>T</td>\n", "      <td>C</td>\n", "      <td>110.9321</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>15.0</td>\n", "      <td>13755.0</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.852457</td>\n", "      <td>0.147543</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>5 rows \u00d7 21 columns</p>\n", "</div>"], "text/plain": ["           Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "5   3.271700e-310  rs11855410    15.0  48593854.0              C           T   \n", "10  3.271700e-310  rs72623972    15.0  48586994.0              C           T   \n", "11  3.271700e-310   rs8026751    15.0  48582845.0              A           G   \n", "12  3.271700e-310   rs7174204    15.0  48579146.0              G           C   \n", "15  3.271700e-310  rs60820695    15.0  48584883.0              T           C   \n", "\n", "      Zscore             Gene GeneSymbol  GeneChr  ...  NrCohorts  NrSamples  \\\n", "5   122.8678  ENSG00000074803    SLC12A1     15.0  ...       15.0    13742.0   \n", "10  115.8559  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "11  115.6258  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "12  115.1496  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "15  110.9321  ENSG00000074803    SLC12A1     15.0  ...       15.0    13755.0   \n", "\n", "    FDR    BonferroniP  ensembl_gene_id start_b37   end_b37 AlleleB  \\\n", "5   0.0  4.166200e-302  ENSG00000074803  48483861  48596275       T   \n", "10  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       T   \n", "11  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       G   \n", "12  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "15  0.0  4.166200e-302  ENSG00000074803  48483861  48596275       C   \n", "\n", "   AlleleB_all     new_f  \n", "5     0.817454  0.182546  \n", "10    0.819253  0.180747  \n", "11    0.819053  0.180947  \n", "12    0.815279  0.184721  \n", "15    0.852457  0.147543  \n", "\n", "[5 rows x 21 columns]"]}, "metadata": {}}]}, {"cell_type": "code", "source": "mergetemp= merge2[~merge2['SNP'].isin(flip['SNP'])]\nmergetemp['new_f'] = mergetemp['AlleleB_all']", "metadata": {}, "execution_count": 66, "outputs": [{"name": "stderr", "output_type": "stream", "text": "/opt/conda/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n\nA value is trying to be set on a copy of a slice from a DataFrame.\n\nTry using .loc[row_indexer,col_indexer] = value instead\n\n\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n\n  \n"}]}, {"cell_type": "code", "source": "mergefinal = mergetemp.append(flip)\nmergefinal.tail()", "metadata": {}, "execution_count": 69, "outputs": [{"execution_count": 69, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>...</th>\n", "      <th>NrCohorts</th>\n", "      <th>NrSamples</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>AlleleB</th>\n", "      <th>AlleleB_all</th>\n", "      <th>new_f</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>96340</th>\n", "      <td>0.000011</td>\n", "      <td>rs75702571</td>\n", "      <td>2.0</td>\n", "      <td>39487073.0</td>\n", "      <td>A</td>\n", "      <td>T</td>\n", "      <td>4.3949</td>\n", "      <td>ENSG00000011566</td>\n", "      <td>MAP4K3</td>\n", "      <td>2.0</td>\n", "      <td>...</td>\n", "      <td>14.0</td>\n", "      <td>8850.0</td>\n", "      <td>0.028673</td>\n", "      <td>1.0</td>\n", "      <td>ENSG00000011566</td>\n", "      <td>39476407</td>\n", "      <td>39664453</td>\n", "      <td>T</td>\n", "      <td>0.827448</td>\n", "      <td>0.172552</td>\n", "    </tr>\n", "    <tr>\n", "      <th>96348</th>\n", "      <td>0.000015</td>\n", "      <td>rs2717701</td>\n", "      <td>16.0</td>\n", "      <td>3066168.0</td>\n", "      <td>C</td>\n", "      <td>G</td>\n", "      <td>-4.3306</td>\n", "      <td>ENSG00000006327</td>\n", "      <td>TNFRSF12A</td>\n", "      <td>16.0</td>\n", "      <td>...</td>\n", "      <td>33.0</td>\n", "      <td>23965.0</td>\n", "      <td>0.037874</td>\n", "      <td>1.0</td>\n", "      <td>ENSG00000006327</td>\n", "      <td>3068446</td>\n", "      <td>3072384</td>\n", "      <td>G</td>\n", "      <td>0.627290</td>\n", "      <td>0.372710</td>\n", "    </tr>\n", "    <tr>\n", "      <th>96349</th>\n", "      <td>0.000016</td>\n", "      <td>rs2257295</td>\n", "      <td>16.0</td>\n", "      <td>3065596.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>-4.3189</td>\n", "      <td>ENSG00000006327</td>\n", "      <td>TNFRSF12A</td>\n", "      <td>16.0</td>\n", "      <td>...</td>\n", "      <td>32.0</td>\n", "      <td>23850.0</td>\n", "      <td>0.039721</td>\n", "      <td>1.0</td>\n", "      <td>ENSG00000006327</td>\n", "      <td>3068446</td>\n", "      <td>3072384</td>\n", "      <td>T</td>\n", "      <td>0.629747</td>\n", "      <td>0.370253</td>\n", "    </tr>\n", "    <tr>\n", "      <th>96350</th>\n", "      <td>0.000002</td>\n", "      <td>rs701929</td>\n", "      <td>1.0</td>\n", "      <td>211846876.0</td>\n", "      <td>G</td>\n", "      <td>A</td>\n", "      <td>4.7223</td>\n", "      <td>ENSG00000117650</td>\n", "      <td>NEK2</td>\n", "      <td>1.0</td>\n", "      <td>...</td>\n", "      <td>35.0</td>\n", "      <td>31352.0</td>\n", "      <td>0.006332</td>\n", "      <td>1.0</td>\n", "      <td>ENSG00000117650</td>\n", "      <td>211836114</td>\n", "      <td>211848960</td>\n", "      <td>A</td>\n", "      <td>0.498415</td>\n", "      <td>0.501585</td>\n", "    </tr>\n", "    <tr>\n", "      <th>96351</th>\n", "      <td>0.000002</td>\n", "      <td>rs697003</td>\n", "      <td>1.0</td>\n", "      <td>211843076.0</td>\n", "      <td>G</td>\n", "      <td>C</td>\n", "      <td>4.7198</td>\n", "      <td>ENSG00000117650</td>\n", "      <td>NEK2</td>\n", "      <td>1.0</td>\n", "      <td>...</td>\n", "      <td>35.0</td>\n", "      <td>31349.0</td>\n", "      <td>0.006385</td>\n", "      <td>1.0</td>\n", "      <td>ENSG00000117650</td>\n", "      <td>211836114</td>\n", "      <td>211848960</td>\n", "      <td>C</td>\n", "      <td>0.497773</td>\n", "      <td>0.502227</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>5 rows \u00d7 21 columns</p>\n", "</div>"], "text/plain": ["         Pvalue         SNP  SNPChr       SNPPos AssessedAllele OtherAllele  \\\n", "96340  0.000011  rs75702571     2.0   39487073.0              A           T   \n", "96348  0.000015   rs2717701    16.0    3066168.0              C           G   \n", "96349  0.000016   rs2257295    16.0    3065596.0              C           T   \n", "96350  0.000002    rs701929     1.0  211846876.0              G           A   \n", "96351  0.000002    rs697003     1.0  211843076.0              G           C   \n", "\n", "       Zscore             Gene GeneSymbol  GeneChr  ...  NrCohorts  NrSamples  \\\n", "96340  4.3949  ENSG00000011566     MAP4K3      2.0  ...       14.0     8850.0   \n", "96348 -4.3306  ENSG00000006327  TNFRSF12A     16.0  ...       33.0    23965.0   \n", "96349 -4.3189  ENSG00000006327  TNFRSF12A     16.0  ...       32.0    23850.0   \n", "96350  4.7223  ENSG00000117650       NEK2      1.0  ...       35.0    31352.0   \n", "96351  4.7198  ENSG00000117650       NEK2      1.0  ...       35.0    31349.0   \n", "\n", "            FDR  BonferroniP  ensembl_gene_id  start_b37    end_b37 AlleleB  \\\n", "96340  0.028673          1.0  ENSG00000011566   39476407   39664453       T   \n", "96348  0.037874          1.0  ENSG00000006327    3068446    3072384       G   \n", "96349  0.039721          1.0  ENSG00000006327    3068446    3072384       T   \n", "96350  0.006332          1.0  ENSG00000117650  211836114  211848960       A   \n", "96351  0.006385          1.0  ENSG00000117650  211836114  211848960       C   \n", "\n", "      AlleleB_all     new_f  \n", "96340    0.827448  0.172552  \n", "96348    0.627290  0.372710  \n", "96349    0.629747  0.370253  \n", "96350    0.498415  0.501585  \n", "96351    0.497773  0.502227  \n", "\n", "[5 rows x 21 columns]"]}, "metadata": {}}]}, {"cell_type": "code", "source": "#Beta = z / sqrt(2p(1\u2212 p)(n + z^2)) and\n#SE =1 / sqrt(2p(1\u2212 p)(n + z^2))\n\nmergefinal['Zscore'] = pd.to_numeric(mergefinal['Zscore'])\nmergefinal['new_f'] = pd.to_numeric(mergefinal['new_f'])\nmergefinal['NrSamples'] = pd.to_numeric(mergefinal['NrSamples'])\n\nmergefinal['beta'] = mergefinal['Zscore'] / np.sqrt(((2 * mergefinal['new_f']) * (1 - mergefinal['new_f'])) * (mergefinal['NrSamples'] + (mergefinal['Zscore']**2)))\n\nmergefinal['se'] = 1 / np.sqrt((2 * mergefinal['new_f']) * (1 - mergefinal['new_f']) * (mergefinal['NrSamples'] + mergefinal['Zscore']**2))\n\n", "metadata": {}, "execution_count": 75, "outputs": []}, {"cell_type": "code", "source": "mergefinal.head()", "metadata": {}, "execution_count": 76, "outputs": [{"execution_count": 76, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>Pvalue</th>\n", "      <th>SNP</th>\n", "      <th>SNPChr</th>\n", "      <th>SNPPos</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Zscore</th>\n", "      <th>Gene</th>\n", "      <th>GeneSymbol</th>\n", "      <th>GeneChr</th>\n", "      <th>...</th>\n", "      <th>FDR</th>\n", "      <th>BonferroniP</th>\n", "      <th>ensembl_gene_id</th>\n", "      <th>start_b37</th>\n", "      <th>end_b37</th>\n", "      <th>AlleleB</th>\n", "      <th>AlleleB_all</th>\n", "      <th>new_f</th>\n", "      <th>beta</th>\n", "      <th>se</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs964611</td>\n", "      <td>15.0</td>\n", "      <td>48597514.0</td>\n", "      <td>A</td>\n", "      <td>C</td>\n", "      <td>193.7198</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>A</td>\n", "      <td>0.156582</td>\n", "      <td>0.156582</td>\n", "      <td>1.664494</td>\n", "      <td>0.008592</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs61248772</td>\n", "      <td>15.0</td>\n", "      <td>48598726.0</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>192.9769</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>T</td>\n", "      <td>0.156809</td>\n", "      <td>0.156809</td>\n", "      <td>1.661796</td>\n", "      <td>0.008611</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs74011998</td>\n", "      <td>15.0</td>\n", "      <td>48596713.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.9409</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156557</td>\n", "      <td>0.156557</td>\n", "      <td>1.662835</td>\n", "      <td>0.008618</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7168752</td>\n", "      <td>15.0</td>\n", "      <td>48600780.0</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>192.7138</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156582</td>\n", "      <td>0.156582</td>\n", "      <td>1.662162</td>\n", "      <td>0.008625</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>3.271700e-310</td>\n", "      <td>rs7162936</td>\n", "      <td>15.0</td>\n", "      <td>48594608.0</td>\n", "      <td>C</td>\n", "      <td>G</td>\n", "      <td>159.4042</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>SLC12A1</td>\n", "      <td>15.0</td>\n", "      <td>...</td>\n", "      <td>0.0</td>\n", "      <td>4.166200e-302</td>\n", "      <td>ENSG00000074803</td>\n", "      <td>48483861</td>\n", "      <td>48596275</td>\n", "      <td>C</td>\n", "      <td>0.156247</td>\n", "      <td>0.156247</td>\n", "      <td>1.657436</td>\n", "      <td>0.010398</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>5 rows \u00d7 23 columns</p>\n", "</div>"], "text/plain": ["          Pvalue         SNP  SNPChr      SNPPos AssessedAllele OtherAllele  \\\n", "0  3.271700e-310    rs964611    15.0  48597514.0              A           C   \n", "1  3.271700e-310  rs61248772    15.0  48598726.0              T           A   \n", "2  3.271700e-310  rs74011998    15.0  48596713.0              C           T   \n", "3  3.271700e-310   rs7168752    15.0  48600780.0              C           T   \n", "4  3.271700e-310   rs7162936    15.0  48594608.0              C           G   \n", "\n", "     Zscore             Gene GeneSymbol  GeneChr  ...  FDR    BonferroniP  \\\n", "0  193.7198  ENSG00000074803    SLC12A1     15.0  ...  0.0  4.166200e-302   \n", "1  192.9769  ENSG00000074803    SLC12A1     15.0  ...  0.0  4.166200e-302   \n", "2  192.9409  ENSG00000074803    SLC12A1     15.0  ...  0.0  4.166200e-302   \n", "3  192.7138  ENSG00000074803    SLC12A1     15.0  ...  0.0  4.166200e-302   \n", "4  159.4042  ENSG00000074803    SLC12A1     15.0  ...  0.0  4.166200e-302   \n", "\n", "   ensembl_gene_id  start_b37   end_b37 AlleleB AlleleB_all     new_f  \\\n", "0  ENSG00000074803   48483861  48596275       A    0.156582  0.156582   \n", "1  ENSG00000074803   48483861  48596275       T    0.156809  0.156809   \n", "2  ENSG00000074803   48483861  48596275       C    0.156557  0.156557   \n", "3  ENSG00000074803   48483861  48596275       C    0.156582  0.156582   \n", "4  ENSG00000074803   48483861  48596275       C    0.156247  0.156247   \n", "\n", "       beta        se  \n", "0  1.664494  0.008592  \n", "1  1.661796  0.008611  \n", "2  1.662835  0.008618  \n", "3  1.662162  0.008625  \n", "4  1.657436  0.010398  \n", "\n", "[5 rows x 23 columns]"]}, "metadata": {}}]}, {"cell_type": "code", "source": "eqtl_5kb_format = mergefinal[['GeneSymbol', 'SNP', 'beta', 'se', 'new_f', 'AssessedAllele', 'OtherAllele', 'Pvalue', 'NrSamples']]\neqtl_5kb_format.head()", "metadata": {}, "execution_count": 80, "outputs": [{"execution_count": 80, "output_type": "execute_result", "data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>GeneSymbol</th>\n", "      <th>SNP</th>\n", "      <th>beta</th>\n", "      <th>se</th>\n", "      <th>new_f</th>\n", "      <th>AssessedAllele</th>\n", "      <th>OtherAllele</th>\n", "      <th>Pvalue</th>\n", "      <th>NrSamples</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>SLC12A1</td>\n", "      <td>rs964611</td>\n", "      <td>1.664494</td>\n", "      <td>0.008592</td>\n", "      <td>0.156582</td>\n", "      <td>A</td>\n", "      <td>C</td>\n", "      <td>3.271700e-310</td>\n", "      <td>13755.0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>SLC12A1</td>\n", "      <td>rs61248772</td>\n", "      <td>1.661796</td>\n", "      <td>0.008611</td>\n", "      <td>0.156809</td>\n", "      <td>T</td>\n", "      <td>A</td>\n", "      <td>3.271700e-310</td>\n", "      <td>13755.0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>SLC12A1</td>\n", "      <td>rs74011998</td>\n", "      <td>1.662835</td>\n", "      <td>0.008618</td>\n", "      <td>0.156557</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>3.271700e-310</td>\n", "      <td>13753.0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>SLC12A1</td>\n", "      <td>rs7168752</td>\n", "      <td>1.662162</td>\n", "      <td>0.008625</td>\n", "      <td>0.156582</td>\n", "      <td>C</td>\n", "      <td>T</td>\n", "      <td>3.271700e-310</td>\n", "      <td>13755.0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>4</th>\n", "      <td>SLC12A1</td>\n", "      <td>rs7162936</td>\n", "      <td>1.657436</td>\n", "      <td>0.010398</td>\n", "      <td>0.156247</td>\n", "      <td>C</td>\n", "      <td>G</td>\n", "      <td>3.271700e-310</td>\n", "      <td>9671.0</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["  GeneSymbol         SNP      beta        se     new_f AssessedAllele  \\\n", "0    SLC12A1    rs964611  1.664494  0.008592  0.156582              A   \n", "1    SLC12A1  rs61248772  1.661796  0.008611  0.156809              T   \n", "2    SLC12A1  rs74011998  1.662835  0.008618  0.156557              C   \n", "3    SLC12A1   rs7168752  1.662162  0.008625  0.156582              C   \n", "4    SLC12A1   rs7162936  1.657436  0.010398  0.156247              C   \n", "\n", "  OtherAllele         Pvalue  NrSamples  \n", "0           C  3.271700e-310    13755.0  \n", "1           A  3.271700e-310    13755.0  \n", "2           T  3.271700e-310    13753.0  \n", "3           T  3.271700e-310    13755.0  \n", "4           G  3.271700e-310     9671.0  "]}, "metadata": {}}]}, {"cell_type": "code", "source": "eqtl_5kb_format.to_csv('eqtl_5kb_format.txt', index=False, sep='\\t')", "metadata": {}, "execution_count": 81, "outputs": []}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": []}]}